### PR TITLE
Added check for `management.contextPath` in Eureka status and health url construction (#1795)

### PR DESCRIPTION
**Pull request by strelok1**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@strelok1 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1823)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1823) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@strelok1 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1823)!

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 1]
```diff
@@ -88,6 +88,9 @@
 	@Value("${management.port:${MANAGEMENT_PORT:${server.port:${SERVER_PORT:${PORT:8080}}}}}")
 	private int managementPort;
 
+	@Value("${management.contextPath:${management.context-path:${MANAGEMENT_CONTEXT_PATH:${server.contextPath:${SERVER_CONTEXT_PATH:${CONTEXT_PATH:}}}}}}")
```
Comment (spencergibb) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: We shouldn't add more of these. We should use the `RelaxedPropertyResolver ` already available.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 2]
```diff
@@ -136,9 +139,9 @@ public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils) {
 			}
 			String scheme = instance.getSecurePortEnabled() ? "https" : "http";
 			instance.setStatusPageUrl(scheme + "://" + instance.getHostname() + ":"
-					+ this.managementPort + instance.getStatusPageUrlPath());
+					+ this.managementPort + this.managementContextPath + instance.getStatusPageUrlPath());
```
Comment (spencergibb) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: Should only add this if not empty (ie not null and not empty). If it were null, `null` would go here.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 3]
```diff
@@ -77,201 +78,201 @@
...
+      // We don't register during bootstrap by default, but there will be another
+      // chance later.
+      client.setRegisterWithEureka(false);
+    }
+    return client;
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(value = EurekaInstanceConfig.class, search = SearchStrategy.CURRENT)
+  public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils) throws MalformedURLException {
```
Comment (strelok1) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: Apologies for the Intellij reformat ... all the changes are still here.

**Comment by spencergibb** [comment]
Looks like you have extra commits? Your history looks wierd now.

**Comment by strelok1** [comment]
@spencergibb i think i did `git pull --rebase`. Should I just start again? Relevant xkcd: https://xkcd.com/1597/

**Comment by spencergibb** [comment]
You rebase from master then force push. Unsure how to clean it up.

**Comment by strelok1** [comment]
Me neither. Stand by for new PR.

**Pull request closed without merging**